[WIP] [FIXED JENKINS-12875] Change default crumb name to Jenkins-Crumb

daniel-beck opened the pull request:
[JENKINS-12875](https://issues.jenkins-ci.org/browse/JENKINS-12875)

This aims to replace #596 based on @jglick's comments there.

Work in progress: Need to add test for the compatibility behavior.

daniel-beck committed:
[FIXED JENKINS-12875] Change default crumb name to Jenkins-Crumb

nthykier commented:
@jglick Hi,

The current apiXml method does not seem to document the original or the new default field name explicitly, as far as I can tell.  That said, I do not mind writing a patch for it if you prefer, as the current default ".crumb" is not usable in our standard setup.

The code is below:

```java
    @PresetData(PresetData.DataSet.ANONYMOUS_READONLY)
    @Test public void apiXml() throws Exception {
        WebClient wc = r.createWebClient();
        r.assertXPathValue(wc.goToXml("crumbIssuer/api/xml"), "//crumbRequestField", r.jenkins.getCrumbIssuer().getCrumbRequestField());
        String text = wc.goTo("crumbIssuer/api/xml?xpath=concat(//crumbRequestField,'=',//crumb)", "text/plain").getWebResponse().getContentAsString();
        assertTrue(text, text.matches("\\Q" + r.jenkins.getCrumbIssuer().getCrumbRequestField() + "\\E=[0-9a-f]+"));
        text = wc.goTo("crumbIssuer/api/xml?xpath=concat(//crumbRequestField,\":\",//crumb)", "text/plain").getWebResponse().getContentAsString();
        assertTrue(text, text.matches("\\Q" + r.jenkins.getCrumbIssuer().getCrumbRequestField() + "\\E:[0-9a-f]+"));
        text = wc.goTo("crumbIssuer/api/xml?xpath=/*/crumbRequestField/text()", "text/plain").getWebResponse().getContentAsString();
        assertEquals(r.jenkins.getCrumbIssuer().getCrumbRequestField(), text);
        text = wc.goTo("crumbIssuer/api/xml?xpath=/*/crumb/text()", "text/plain").getWebResponse().getContentAsString();
        assertTrue(text, text.matches("[0-9a-f]+"));
        wc.assertFails("crumbIssuer/api/xml?xpath=concat('hack=\"',//crumb,'\"')", HttpURLConnection.HTTP_FORBIDDEN);
        wc.assertFails("crumbIssuer/api/xml?xpath=concat(\"hack='\",//crumb,\"'\")", HttpURLConnection.HTTP_FORBIDDEN);
        wc.assertFails("crumbIssuer/api/xml?xpath=concat('{',//crumb,':1}')", HttpURLConnection.HTTP_FORBIDDEN); // 37.5% chance that crumb ~ /[a-f].+/
        wc.assertFails("crumbIssuer/api/xml?xpath=concat('hack.',//crumb,'=1')", HttpURLConnection.HTTP_FORBIDDEN); // ditto
        r.jenkins.getCrumbIssuer().getDescriptor().setCrumbRequestField("_crumb");
        wc.assertFails("crumbIssuer/api/xml?xpath=concat(//crumbRequestField,'=',//crumb)", HttpURLConnection.HTTP_FORBIDDEN); // perhaps interpretable as JS number
    }
```

jglick commented:
I meant that `apiXml` continues to pass with the patch. The diff of `CrumbIssuer.java` looks like it should make it work.

tfennelly committed:
Revert "Merge pull request #2106 from gusreiber/2.0_j-33411" (scrollspy changes)

This reverts commit 9fb542369206c8abb886360cb2bb23d06db86493, reversing
changes made to 36528a681a541b21bfd4d4861fcae91c76df8385.

tfennelly committed:
Fix select.options empty check

tfennelly committed:
Re-fix jshint errors

kzantow committed:
JENKINS-33296 - plugin dependency issues during install

recena committed:
[JENKINS-31162] Category configuration reviewed

tfennelly committed:
Refactor config-tabbar.js, extracting common bits to tabbar.js

This is in preparation for readding scrollspy

tfennelly committed:
Move button repositioning fire event for finder input onfocus

Commit:
Ensure the "Getting Started" dialogue title is consistent between views

The different views are distinguished by the header (h1) tags, for example
"Customize Jenkins" or "Jenkins is ready!" and I find it jarring to have the
dialogue title change.

The context of the whole series of views in the dialogue is "Getting Started" so
I think the dialogue title should reflect that scoping of actions.

tfennelly committed:
Added config-scrollspy.js bundle

And changed Job config ti use it. It's really just regular tabs atm

tfennelly committed:
Rename ConfigTableMetaData.deactivateActiveSection to ConfigTableMetaData.hideSection

tfennelly committed:
Tab section show/hide via CSS Vs JavaScript

This will allow us to apply different behaviour for scrollspy show/hide, leaving the common JavaScript the same

tfennelly committed:
Tabbar & Scrollspy specific LESS files

tfennelly committed:
Changed config table CSS class to "config-table"

It was "job-config".

tfennelly committed:
Added config-table-top-row class to the top level <tr>s in the config table

tfennelly committed:
Enable scrollspy

tfennelly committed:
Applying styles

tfennelly committed:
Fix Save/Apply All buttons

tfennelly committed:
behavior-loading styling

Commit:
[JENKINS-31162] @jglick's comment was addressed. A getEffectiveClazzFor() method was added to determine effective clazz of CustomOrganizationFolderDescriptor

tfennelly committed:
Hide left panel and center main panel

Commit:
[JENKINS-31162] A CopyCategory is returned when it makes sense

kzantow committed:
JENKINS-33308 - automatically install previously bundled dependencies
for plugins that depend on old Jenkins versions

tfennelly committed:
Fix highlight text removal

tfennelly committed:
Renaming tabbed config spec html file

kohsuke committed:
Merge pull request #2123 from rtyler/2.0-consistent-titles

Ensure the "Getting Started" dialogue title is consistent between views

Commit:
[JENKINS-31162] iconClassName field has been remove from Category entity. name field has been renamed to displayName

kzantow committed:
Address @jglick's comments: refactor ClassicPluginStrategy to use
BREAK_CYCLE, use the same getImpliedDependency method for multiple code
paths, @Restricted isMailerPresent

Commit:
[JENKINS-31162] Removed CopyCategory

tfennelly committed:
Tweaking tabbar-spec tests

kzantow committed:
Implement @jglick's idea: move implicit dependency modifications to
UpdateSite, when reading plugin metadata

kzantow committed:
Refactoring lack-of-moving-comments-too fail

Commit:
fixing for 'displayName'...also gulpfule in

Commit:
reconcieling new file names shared by addItem and config tabs

Commit:
styling

Commit:
restoring images

kzantow committed:
JENKINS-33572 - initial admin user should not retain setup token

recena committed:
[JENKINS-31162] A new category (UncategorizedCategory) has been added

Commit:
triming out old form elements

recena committed:
[JENKINS-31162] Javadoc mistake

kzantow committed:
Need to save the user...

Commit:
removing 'all' from config buttons

Commit:
adding title text for copy item

Commit:
removing  per code review | cleaning up styles

Commit:
[JENKINS-31162] A new API was added to handle Item icons

tfennelly committed:
Fixing more scrollspy bugs + adding some tests

Commit:
Merge remote-tracking branch 'upstream/2.0' into JENKINS-31162

tfennelly committed:
Fix rowGroupContainer undefined bug

tfennelly committed:
Hide the finder widget for scrollspy

As requested by DanielB and Tyler

tfennelly committed:
Hide tabs for invisible sections

daniel-beck committed:
Merge branch 'master' of github.com:jenkinsci/jenkins into 2.0

Conflicts:
	changelog.html
	cli/pom.xml
	core/pom.xml
	core/src/main/java/hudson/ProxyConfiguration.java
	core/src/main/java/jenkins/model/Jenkins.java
	core/src/main/java/jenkins/slaves/JnlpSlaveHandshake.java
	core/src/main/resources/hudson/model/Messages_da.properties
	pom.xml
	test/pom.xml
	war/pom.xml

Commit:
refining mega screensizing | setting first tab to active as default

daniel-beck committed:
[FIX JENKINS-33571] Remove Slack Plugin from the list

daniel-beck committed:
Merge pull request #2127 from kzantow/JENKINS-33572

[JENKINS-33572] - initial admin user should not retain setup token

recena committed:
Merge pull request #1 from gusreiber/M-31162

These changes providesthe front-end part of JENKINS-31162.

recena committed:
[JENKINS-31162] Some reviews to @gusreiber's PR

recena committed:
Merge branch 'JENKINS-31162' of https://github.com/recena/jenkins into JENKINS-31162

daniel-beck committed:
Merge branch '2.0' of github.com:jenkinsci/jenkins into 2.0

daniel-beck committed:
Merge pull request #2122 from kzantow/2.0

[JENKINS-33296] - plugin dependency issues during install

daniel-beck committed:
Merge pull request #2134 from daniel-beck/JENKINS-33571

[FIX JENKINS-33571] Remove Slack Plugin from the list

tfennelly committed:
Replace Behaviour.specify with a homegrown jQuery alt

Behaviour.specify is randomly failing

tfennelly committed:
Scan for missing sections and hide tabs if needed

tfennelly committed:
Ignore invisible/hidden sections in watchScroll

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.0-alpha-4

Commit:
default item styling

kzantow committed:
JENKINS-33599 - write initial admin password to a file in the Jenkins
home directory

Commit:
blocking submit button after click to avoid double submit

Commit:
fixing jerky redraw

Commit:
more fixes to jerkiness

recena committed:
Merge pull request #2 from gusreiber/rec-31162

Reviews in the UI.

tfennelly committed:
Only track section visibility (and hide tabs) for scrollspy

kzantow committed:
Merge remote-tracking branch 'primary/2.0' into JENKINS-33599-write-file-for-security-token

kzantow committed:
Add @since

kzantow committed:
[JENKINS-33534] - timing issue casuing installer plugin status
indicators not to appear & fix installingPlugin js data structure

daniel-beck committed:
Merge pull request #2135 from kzantow/JENKINS-33534-missing-setup-indicators

[JENKINS-33534] - missing installer plugin status indicators

tfennelly committed:
ConfigTableMetaData.trackSectionVisibility to now show (as well as hide) activators for sections that appear

Commit:
first page clean-up

Commit:
Merge pull request #3 from gusreiber/rec-31162

first page clean-up

Commit:
show-hide menu|1st page|bug fixes

Commit:
Merge pull request #4 from gusreiber/rec-31162

show-hide menu|1st page|bug fixes

tfennelly committed:
Split watchScroll into 2 scroll listeners, one or auto tab activation and the other for sticking the tabbar to the top of the window.

amuniz committed:
Merge branch 'JENKINS-14538' of github.com:daniel-beck/jenkins into JENKINS-14538

tfennelly committed:
Applying @kzantow fix for the draggable sections regression

Thanks @kzantow and sorry for missing this!!

Commit:
[JENKINS-31162] Big review of the current approach

tfennelly committed:
Merge pull request #2121 from tfennelly/JENKINS-33495

[JENKINS-33495 and JENKINS-33496] Separate scrollspy widget code out from tab widget code

kohsuke committed:
[JENKINS-33662] Implemented a quick & dirty upgrade wizard

kohsuke committed:
Oops, no wonder why breadcrumb gets double height

kohsuke committed:
i18n

kohsuke committed:
New installation should be set to 2.0 level

kohsuke committed:
Need to update the local state to reflect the current level change

kohsuke committed:
Set the file permission to 0640

Note that we already go much beyond what normal webapps do, which is to
show the initial configuration dialog unauthenticated to anyone with
access.

kohsuke committed:
No need for plugins to access

kohsuke committed:
incorrect usage of BulkChange

kohsuke committed:
No need to take those parameters when unused

In fact it's desirable not to have them to keep things more POJO

kohsuke committed:
Save the lookup

and its dataflow analysis warning

kohsuke committed:
Put this into the secrets dir

... so that the nature of this file is little more clearer

kohsuke committed:
Merge pull request #2142 from jenkinsci/PR-2138

[FIXED JENKINS-33599] write file with admin password for installer

amuniz committed:
Save tools configuration after configure

amuniz committed:
Fix Maven tool configuration test

amuniz committed:
Fix JDK tool installer test

amuniz committed:
Moved to jenkins.tools package

amuniz committed:
Reuse local variable

amuniz committed:
JDKs were not loaded from disk

amuniz committed:
API use restricted and simplified logic in Functions

Commit:
[JENKINS-31162] Merge from upstream/2.0 and resolve conflicts

kohsuke committed:
Permission check was missing

kohsuke committed:
Merge remote-tracking branch 'origin/2.0' into JENKINS-33662

Conflicts:
	core/src/main/java/jenkins/install/SetupWizard.java

kohsuke committed:
Resolved a merge failure

Commit:
adding missing less file

Commit:
[JENKINS-31162] @jglick's comments were addressed (partially)

Commit:
[JENKINS-31162] Another @jglick's comments were addressed

Commit:
[JENKINS-31162] Another (part III) @jglick's comments were addressed

Commit:
[JENKINS-31162] @glick's comments were addressed

recena committed:
[JENKINS-31162] Constants as identifiers of the categories

recena committed:
[JENKINS-31162] More context in a javadoc

recena committed:
[JENKINS-31162] ItemCategory no londer implements ModelObject

daniel-beck committed:
[FIX JENKINS-33705] Use platform line endings

daniel-beck committed:
[FIX JENKINS-33706] Remove ineligible build-monitor-plugin

recena committed:
[JENKINS-31162] Changed the modifier of setWeight(int weight) method

daniel-beck committed:
Refine the plugin selection

daniel-beck committed:
Add Slack Plugin again, 2.0.1 resolved cause for JENKINS-33571

recena committed:
[JENKINS-31162] LOGGER.log() instead of LOGGER.warning()

Commit:
merging new item and config item|splitting less files

Commit:
woops reverted newList/form to wrong file..

Commit:
cleaning page centering for config

Commit:
fixing copy item check

kohsuke committed:
Merge pull request #2141 from jenkinsci/JENKINS-33662

[FIXED JENKINS-33662] upgrade wizard step #0

Commit:
error handling item selection

Commit:
coloring default icons

kzantow committed:
JENKINS-33659 - Basic fix for scrollspy width issues

daniel-beck committed:
Remove Slack Plugin again.

daniel-beck committed:
Merge pull request #2151 from daniel-beck/JENKINS-33706

[FIX JENKINS-33706] Remove ineligible build-monitor-plugin

recena committed:
Merge pull request #9 from gusreiber/JENKINS-33713

Gus's work focused on the UI

Commit:
[JENKINS-31162] The service is now under View instead of View + ItemGroupMixin

kzantow committed:
JENKINS-33557 - tcp timeouts cause installer to hang for 6 minutes

Commit:
[JENKINS-31162] Removed a @since annotation in javadoc

Commit:
[JENKINS-31162] Removing not illuminated javadoc

Commit:
[JENKINS-31162] More reasonable ordinal value for StandaloneProjectsCategory

kzantow committed:
Mask security token during setup wizard

jglick committed:
Do not show f:breadcrumb-config-outline for configuration pages with tabs.

Commit:
[JENKINS-31162] Removed an unused jelly var. Additionally, caused a NPE

Commit:
[JENKINS-31162] Unrelated change

Commit:
[JENKINS-31162] Wrong attribute in <l:layout>

jglick committed:
Merge pull request #2158 from jglick/no-breadcrumb-config-outline

Do not show f:breadcrumb-config-outline for configuration pages with tabs

Commit:
[JENKINS-31162] Removed unrelated change

jglick committed:
Merge pull request #2157 from kzantow/mask-security-key

Mask security token during setup wizard

Commit:
[JENKINS-31162] Unnecessary definition of data-root as HTML attribute

kzantow committed:
JENKINS-33246 - add a footnote about required dependencies installing
during setup wizard

Commit:
fixing radio reset bug

recena committed:
Merge pull request #14 from gusreiber/jmr-31162

Fixing radio reset bug

Commit:
[JENKINS-31162] cssclass instead of cssClass

Commit:
[JENKINS-31162] API REST method instead of API method

Commit:
[JENKINS-31162] Removed parts not used in newJob.jelly

Commit:
[JENKINS-31162] Another occurence of cssClass instead of cssclass

kzantow committed:
Darker dependency text, increase contrast

Commit:
relative possition for restapi | cssclass attribute fix | fix to order

Commit:
need to keep a checkReady

Commit:
Merge branch 'JENKINS-31162' of github.com:recena/jenkins into jmr-31162

kzantow committed:
Fix test for connectivity check

Commit:
remoing unused root param

Commit:
remoing cover dom lint from form check

recena committed:
Merge pull request #15 from gusreiber/jmr-31162

More reviews from @gusreiber

daniel-beck committed:
Merge pull request #2159 from kzantow/JENKINS-33246-show-install-dependency-legend

[FIX JENKINS-33246] Add a footnote about ** in setup wizard

Commit:
fixing prop vs. attr browser bug on OK button

daniel-beck committed:
Merge pull request #2140 from amuniz/JENKINS-14538

[FIX JENKINS-14538] Move tool configuration to separate page

daniel-beck committed:
Merge pull request #2150 from daniel-beck/JENKINS-33705

[FIX JENKINS-33705] Use platform line endings

daniel-beck committed:
AnsiColor Plugin is not actually released from the jenkinsci fork

daniel-beck committed:
Merge pull request #2152 from daniel-beck/plugin-changes

Refine the plugin selection

daniel-beck committed:
Merge pull request #2153 from kzantow/JENKINS-33659-scrollspy-ux-issues

[FIX JENKINS-33659] Basic fix for scrollspy width issues

recena committed:
Merge pull request #16 from gusreiber/pre-31162

Fixed an issue (OK button) with Firefox.

Commit:
Merge remote-tracking branch 'upstream/2.0' into JENKINS-31162

tfennelly committed:
Fix lint errors

tfennelly committed:
Fix lint errors

recena committed:
Merge pull request #17 from tfennelly/recena-JENKINS-31162-PR01

Fix lint errors

tfennelly committed:
Add Item specific LESS file (add-item.less)

tfennelly committed:
Add Item specific LESS file (add-item.less)

tfennelly committed:
Remove jenkins-js-widgets.less

Was readded in error

tfennelly committed:
Remove jenkins-js-widgets.less

Was readded in error

recena committed:
Merge pull request #18 from tfennelly/recena-JENKINS-31162-PR02

Add a specific LESS file (add-item.less)

tfennelly committed:
Remove sidepanel from newJob.jelly

And set the body element classes via the cssclass attr

Commit:
[JENKINS-31162] Fixed how FreeStyleProject returns its IconFilePathPattern

Commit:
Merge branch 'JENKINS-31162' of https://github.com/recena/jenkins into JENKINS-31162

Commit:
[JENKINS-31162] Removed unused import

Commit:
Merge branch 'JENKINS-31162' of https://github.com/recena/jenkins into JENKINS-31162

tfennelly committed:
Fix new item page loading weirdness.

recena committed:
Merge pull request #19 from tfennelly/recena-JENKINS-31162-PR03

Fix new item page loading weirdness

tfennelly committed:
Add "navBox" earlier and wire in.

tfennelly committed:
Check if we can enable the ok button on keyup event (Vs change)

Using the change event resulted in a strange situation where, if the user selected the job type first and thn entered the name, they'd have to tab out of the name entry field before the ok button would be enabled.

recena committed:
Merge pull request #20 from tfennelly/recena-JENKINS-31162-PR04

Check if we can enable the ok button on keyup event (Vs change)

kzantow committed:
Merge remote-tracking branch 'primary/2.0' into JENKINS-33557-tcp-timeouts-during-install

tfennelly committed:
Fix "weight" based sorting algo

recena committed:
Merge pull request #21 from tfennelly/recena-JENKINS-31162-PR05

Fix "weight" based sorting algo

tfennelly committed:
Ignore window scroll events while a manual scroll (tab click) is happening

tfennelly committed:
Ignore the window scroll event that's generated from the repositioning of the new item page tabbar.

Repositioning of the tabbar causes a scroll event to happen, which causes tab activation/reactivation, which in turn causes a further autoscroll etc etc. Is the cause of the weird scrolling behavior as seen by @daniel-beck.

tfennelly committed:
Fixed a comment typo

recena committed:
Merge pull request #22 from tfennelly/recena-JENKINS-31162-PR06

Ignore the window scroll event that's generated from the repositioning of the new item page tabbar

Commit:
[JENKINS-31162] A different way to do the same. Asked by @jglick

kzantow committed:
Modify default ProxyConfiguration.open() to use 20 second connect
timeout, this may be modified with the configuration property:
jenkins.proxy.default_connect_timeout_millis, a value of 0 will restore
the previous behavior

kzantow committed:
JENKINS-33728 - Remove stray getActiveInstance call in layout.jelly

daniel-beck committed:
Merge pull request #2164 from kzantow/JENKINS-33728-dont-call-getactiveinstance-in-layoutjelly

[JENKINS-33728] - Remove stray getActiveInstance call in layout.jelly

Commit:
setting tabs min from gt 1 to gt 3

tfennelly committed:
Fix more funky scrolling + sticky tabbar issues

recena committed:
Merge pull request #23 from gusreiber/k-31162

Minimum number of tabs to show.

recena committed:
Merge pull request #24 from tfennelly/recena-JENKINS-31162-PR07

Fix more funky scrolling + sticky tabbar issues

daniel-beck committed:
Merge branch 'master' of github.com:jenkinsci/jenkins into 2.0

Conflicts:
	cli/pom.xml
	core/pom.xml
	pom.xml
	test/pom.xml
	war/pom.xml

kzantow committed:
Use standard property naming convention

kzantow committed:
Fix scroll on click bug

kzantow committed:
Use ProxyConfiguration defaults in UpdateCenter

tfennelly committed:
Fix more funky scrolling + sticky tabbar issues (missed an 5px offset)

tfennelly committed:
Resize new item page tabbar based on form resize.

recena committed:
Merge pull request #26 from tfennelly/recena-JENKINS-31162-PR08

Resize new item page tabbar based on form resize

Commit:
[JENKINS-31162] Reviewed the ordening applied

Commit:
Merge branch 'JENKINS-31162' of https://github.com/recena/jenkins into JENKINS-31162

Commit:
[JENKINS-31162] Revert an unnecessary change

tfennelly committed:
Sort the items within each category

recena committed:
Merge pull request #25 from kzantow/JENKINS-31162

Fix scroll on click bug

recena committed:
Merge pull request #27 from tfennelly/recena-JENKINS-31162-PR09

Sort the items within each category

tfennelly committed:
Put focus back on item name input after item selection if name is empty.

The item name entry is given focus at the start, but loses it of the user selects an item. This changes puts focus back (post item selection) on the name input if it's empty.

recena committed:
Merge pull request #28 from tfennelly/recena-JENKINS-31162-PR10

Put focus back on item name input after item selection if name is empty

Commit:
[JENKINS-31162] Renamed weight instead of order. Client-side ordering was removed, no needed

batmat committed:
More concise welcome text for Jenkins 2

daniel-beck committed:
Merge pull request #2156 from kzantow/JENKINS-33557-tcp-timeouts-during-install

[JENKINS-33557] - connectivity checks may cause setup wizard to hang for a long time

Commit:
[JENKINS-31162] New special render for flat categories

kohsuke committed:
Only show the upgrade banner when Jenkins is in charge

daniel-beck committed:
Merge pull request #2165 from batmat/install-text-leaner

More concise welcome text for Jenkins 2

daniel-beck committed:
Merge pull request #2117 from recena/JENKINS-31162

[JENKINS-31162] New item categorization

kohsuke committed:
Fixed up tests

Test code needs to now execute in the request scope for us to be able to
determine if Jenkins object is in charge or not

kohsuke committed:
Merge pull request #2166 from jenkinsci/upgrade-wizard

[FIX JENKINS-33767] Only show the upgrade banner when Jenkins is in charge

kohsuke committed:
Next stop, beta 1

kohsuke committed:
Disabling HelpLinkTest

as an emergency measure for shipping 2.0 beta 1

See the comment for why. Probably related to scrollspy

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.0-beta-1

jglick committed:
Incorrect handling of relative paths from createItem, especially from a folder.

jglick committed:
Make ViewDescriptor.doAutoCompleteCopyNewItemFrom honor DirectlyModifiableTopLevelItemGroup.canAdd.

aheritier committed:
Fix missing space in log message

kohsuke committed:
[FIXED JENKINS-33799]

Fix the width & height of the icon via CSS.

Previously, we were relying on the actual image being served as a
correct size. This is fine for icons we ship in Jenkins, but when
plugins use avatars as icons, those images might not come in the correct
size.

oleg-nenashev committed:
Merge pull request #2174 from aheritier/patch-2

Fix missing space in log message

daniel-beck committed:
Merge branch 'master' of github.com:jenkinsci/jenkins into 2.0

Conflicts:
	cli/pom.xml
	core/pom.xml
	pom.xml
	test/pom.xml
	war/pom.xml

jglick committed:
Merge branch '2.0' into JENKINS-33759-misc

jglick committed:
newJob link should be offered according to Item.CREATE permission on the ViewGroup (e.g., Folder), not necessarily root (Jenkins).

jglick committed:
Moving DirectlyModifiableTopLevelItemGroup.canAdd check to a higher UI level, in ViewDescriptor.doAutoCompleteCopyNewItemFrom.

jglick commented:
What is still missing here?

kohsuke committed:
Whatever that makes other people comfortable

jglick committed:
[FIXED JENKINS-33772] ID was not sufficiently unique for SezpozModule; use an ID based on the actual annotated element.

kohsuke committed:
[JENKINS-33799]

As suggested by Tom, removing the tag name restriction
and correcting the xlg class name.

See the Icon class for more details

jglick committed:
Merge pull request #2173 from jglick/JENKINS-33759-misc

Copy item in folder fixes

kohsuke committed:
Merge pull request #2175 from kohsuke/JENKINS-33799

[FIX JENKINS-33799] Enforce correct icon size in list view

jglick committed:
[FIXED JENKINS-20509] Move doCheckJobName to View so that it can be used also from folders.

jglick committed:
Merge pull request #2183 from jglick/SezpozModule-unique-ID-JENKINS-33772

[JENKINS-33772] ID was not sufficiently unique for SezpozModule

jglick committed:
Merge pull request #2192 from jglick/checkJobName-JENKINS-20509

[JENKINS-20509] Move doCheckJobName to View
